app/Listeners/TaskUpdateXP.php

```diff
@@ -26,66 +26,64 @@ public function __construct()
     /**
      * Handle the event.
      *
-     * @param  TaskUpdate $event
+     * @param  ModelUpdate $event
      * @return void
      */
-    public function handle(TaskUpdate $event)
+    public function handle(ModelUpdate $event)
     {
-        // I don't always make PRs, but when I do I make one with every code change I have, even work in progress
-        return $this;
-        $this->tasks = $event->tasks;
+        $this->model = $event->model;
 
         //task user id
-        $user_id = $this->tasks->task_history[0]['user'];
+        $userId = $this->model->task_history[0]['user'];
 
         //project owner id
-        $project_owner_id = $this->tasks->owner;
+        $projectOwnerId = $this->model->ownerId;
 
-        //get task's XP value
-        $task_xp = $this->tasks->xp;
+        $userProfile = Profile::find($userId);
+        $ownerProfile = Profile::find($projectOwnerId);
 
-        //
-        $user_profile = Profile::find($user_id);
-        var_dump($user_profile);
-        $owner_profile = Profile::find($project_owner_id);
 
-        $end = 0;
+        //get task's XP value
+        $taskXp = $this->model->xp;
+
+        $work = 0;
         $review = 0;
-        for ($i = 0; $i < count($this->tasks->task_history); $i++) {
-            if ($this->tasks->submitted_for_qa === true) {
-                $end += $this->tasks->task_history[$i]['timestamp'] - $this->tasks->task_history[$i - 1]['timestamp'];
-                echo $end;
-            } elseif ($this->tasks->submitted_for_qa === false) {
-                $review += $this->tasks->task_history[$i]['timestamp'] - $this->tasks->task_history[$i - 1]['timestamp'];
+        for ($i = count($this->model->task_history) - 1; $i >= 0; $i--) {
+            if (((($this->model->task_history[$i]['event'] == "Task returned for development")) || $this->model->task_history[$i]['event'] == "Task passed QA!") && ($i > 0)) {
+                    $review += floor($this->model->task_history[$i]['timestamp'] / 1000) - floor($this->model->task_history[$i - 1]['timestamp'] / 1000);
```

Comment:
Wrong logic, you can not assume previous record is correct one, you should check if that's correct.